### PR TITLE
Add legacy decorator

### LINKED ISSUE #11087: MAINT: Add category of discouraged but non-deprecated functions
Increasingly it seems like we are keeping old APIs available by keeping tiny wrappers around new, better APIs. To give an example from https://github.com/mne-tools/mne-python/pull/10184, `epochs.plot_psd_topomap(...)` is kept around to avoid breaking code, but we really want people to use `EpochsSpectrum` methods.

I propose we add a new `@deprecated`-like decorator that marks such code as... something. Maybe "legacy" is a good term? And instead of the `.. warning::` that `@deprecated` produces, it could be a `.. info::` or so that says something like:

> This function is provided to support legacy code. New code should use the more complete API provided by {0}

### PR DESCRIPTION
**Pull request by drammock**
closes #11087

### CODE REVIEW

File: mne/utils/docs.py [hunk 1]
```diff
@@ -4041,29 +4040,13 @@ def open_docs(kind=None, version=None):
     webbrowser.open_new_tab('https://mne.tools/%s/%s' % (version, kind))
 
 
-# Following deprecated class copied from scikit-learn
-
-# force show of DeprecationWarning even on python 2.7
-warnings.filterwarnings('always', category=DeprecationWarning, module='mne')
```
Comment (larsoner) [inline mne/utils/docs.py:4047]: Please revert, IIRC this is critical for helping our users see our warnings

**Review by larsoner — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: mne/utils/docs.py (hunk 1 above)
Comment (hoechenberger) [reply mne/utils/docs.py:4047]: Python silences DeprecationWarnings when running in user code. 

It is therefore recommended to use FutureWarnings if we want to ensure that users see the warnings by default. 

We could keep this change here and change our Deprecation- to FutureWarnings

Comment (drammock) [reply mne/utils/docs.py:4047]: Yes, I actually just learned a few days ago that FutureWarnings are for users and DeprecationWarnings are for downstream devs. But I didn't make the connection to this bit of code! @larsoner WDYT about switching our warning type?

Comment (larsoner) [reply mne/utils/docs.py:4047]: Works for me. Either revert this line for now and make a new PR for FutureWarning, or change to `FutureWarning` in this PR and then we can merge

File: mne/utils/docs.py [hunk 2]
```diff
@@ -4124,11 +4096,40 @@ def _update_doc(self, olddoc):
                 if li > 0 and len(line.strip()):
                     n_space = len(line) - len(line.lstrip())
                     break
-            newdoc = "%s\n\n%s%s" % (newdoc, ' ' * n_space, olddoc)
-
+            newdoc = f"{newdoc}\n\n{' ' * n_space}{olddoc}"
         return newdoc
 
 
+# Following deprecated class copied from scikit-learn
```
Comment (drammock) [inline mne/utils/docs.py:4103]:
```suggestion
# make DeprecationWarnings actually show up for users
warnings.filterwarnings('always', category=DeprecationWarning, module='mne')

# Following deprecated class copied from scikit-learn
```

File: mne/utils/docs.py [hunk 3]
```diff
@@ -4041,29 +4040,13 @@ def open_docs(kind=None, version=None):
...
-    arguments populate properly following our verbose decorator methods based
-    on decorator.
-
-    Parameters
-    ----------
-    extra : str
-        Extra information beyond just saying the class/function/method
-        is deprecated.
-    """
+class _decorator:
```
Comment (drammock) [inline mne/utils/docs.py:4043]:
```suggestion
# make DeprecationWarnings actually show up for users
warnings.filterwarnings('always', category=DeprecationWarning, module='mne')


class _decorator:
```

**Review by larsoner — APPROVED** [review]

**Pull request merged**